metaflow/plugins/aws/step_functions/step_functions_cli.py

```diff
@@ -158,19 +164,34 @@ def attach_prefix(name):
       if SFN_STATE_MACHINE_PREFIX is not None:
           return SFN_STATE_MACHINE_PREFIX + '_' + name
       return name
+    project = current.get('project_name')
+    if project:
+        if name:
+            raise MetaflowException("--name is not supported for @projects. "
+                                    "Use --branch instead.")
+        state_machine_name = attach_prefix(current.project_flow_name)
+        project_branch = to_bytes('.'.join((project, current.branch_name)))
+        token_prefix = 'mfprj-%s' % sha1(project_branch).hexdigest()
```

Comment:
why use a hash here? could we just use `%project.%branch` so it is human-readable?

Reply:
good point - the rationale is that `project` & `branch` could each be up to 128 characters - which increases the length of the token prefix.